_topics/en/adaptor-signatures.md

```diff
@@ -0,0 +1,242 @@
+
+      s = r + t + H(P || R + T || m) * p
+
+  She removes `t` from the final signature to produce an adaptor
+  signature:
+
+      s' = s - t
+
+  She gives Bob the adaptor signature, which consists of the following
+  data:
+
+      s', R, T
+
+  Bob can verify the adaptor signature:
+
+      s' * G ?= R + H(P || R+T || m) * P
+
+  But it's not a valid signature.  For a valid signature, BIP340 expects
+  `x` and `Y`, using them with the expression:
+
+      x * G ?= Y + H(P || Y || m) * P
+
+  However,
+
+  - If Bob sets `Y=R` so that it matches the `s'` he received in the
+    adaptor signature, then BIP340 is going to fail on `H(P || R || m)`
+    since Alice computed her hash with `H(P || R + T || m)`.
+
+  - If Bob sets `Y=R + T` so that it matches `H(P || R + T || m)`, BIP340
+    is going to fail on the initial `Y` since Bob is providing `R + T`
+    rather than the needed `R`.
+
+  Therefore Bob can't use the adaptor signature as a final signature.
+  However, he can create his own signature using it.  This is similar to the
+  signature Alice created but Bob doesn't commit to `t` here, since Bob
+  doesn't know that value.  All variables here except `T` are different
+  for Bob than they were for Alice:
+
+      s = r + H(P || R + T || m) * p
+
+  Unlike Alice, Bob doesn't need to tweak his signature.  Bob's `s` is
+  not a valid signature because it commits to `r` and `R + T`, which
+  won't pass BIP340 verification for the same reasons previously described.
+  To be valid, the signature needs to commit to `r + t` and `R + T`,
+  which Bob can't produce since he doesn't know `t`.
+
+  Bob gives Alice his adaptor signature:
+
+      s, R, T
+
+  Alice already knew `T`, but `(s, R, T)` is a standard adaptor
+  signature so we use its full form.  Alice can produce a final
+  signature from that adaptor signature using the hidden `t` value that
+  only she knows so far:
+
+      (s + t) * G ?= R + T + H(P || R + T || m) * P
+
+  Alice uses the final signature to broadcast Bob's transaction that
+  pays her.  When Bob sees `(s + t)` onchain, he can learn the value of `t`:
+
+      t = (s + t) - s
+
+  He can then use `t` to solve the adaptor signature Alice gave him
+  earlier:
+
+      (s' + t) * G ?= R + T + H(P || R + T || m) * P
+
+  Bob uses that final signature to broadcast the transaction Alice
+  originally gave him.
+  </details><br>
+  </div>
+
+  ### Relationship to multiparty signatures
+
+  Adaptor signatures usually can't secure a contract entirely by
+  themselves.  For example, in the above description of a coinswap,
+  Alice could double spend her payment to Bob after she learned Bob's
+  final signature, or Bob could've tried the same in reverse (with more
+  difficulty since we assumed Alice's transaction had one confirmation).
+  This is typically addressed by combining adaptor signatures with
+  multiparty signatures.  For example, Alice deposits her money into an
```

Comment:
I've been using "multiparty signature" to distinguish collaboratively-generated signatures (e.g. MuSig) from the independently-generated signatures glued together with script (e.g. `OP_CHECKMULTISIG`) that we tend to call "multisig" in Bitcoin.  In that context, is this suggestion a minor terminology problem, or do you think it's likely to lead to misunderstadings?

Reply:
That makes sense. I don't think it will lead to misunderstandings. It's just a mess that we have settled on so many terms that are used differently by different communities, and that are not often not very well defined or used imprecisely. I believe in particular the latter is one of the main reasons why it's hard to read technical texts in the Bitcoin space.